### DISCUSSION TITLE
skip_dir for directory containing file type

### DISCUSSION THREAD

**Discussion by NordicAngel** [abraunegg/onedrive#2189] [Q&A] [answered]
I have a few c# projects stored in my onedrive which I don't want synced, but I do want the other the stuff in the folder to sync.
Let's say I have a directory with a solution file (.sln)
`\Code\Example_Project\example.sln`
Could I skip the Example_Project directory (or at least skip all the content of Example_Project) but sync the rest of Code directory?
This sorting should be based on the file type .sln as that is the only common denominator I am aware of.

**Comment by abraunegg** [comment, marked as answer]
@NordicAngel 
> Could I skip the Example_Project directory (or at least skip all the content of Example_Project) but sync the rest of Code directory?

Yes you can - please read the usage document in regards to 'skip_dir' and 'skip_file'

**Reply by NordicAngel** [reply]
I can't find anywhere it says how to skip a directory that has a specific file. I can find skip directory or skip file, but not skip directory based on file (or file type in this case).

**Comment by abraunegg** [comment]
@NordicAngel 
'skip_dir' configurations are for skipping directories, however you can use wildcards to cover greater skip items. Files are not skipped, only directories are applied with this directive. You cannot skip a directory containing a specific file using this option, but you can skip the parent directory - which, would skip that file .......


'skip_file' configurations are for skipping files, however you can use wildcards to cover greater skip items. Directories are not skipped, only files are applied with this directive.


You can also look at 'sync_list' to build up a folder structure of what files and directories you want to include to sync, but then also include specific exclusions as well.
